parser.cpp

```diff
@@ -8,77 +6,65 @@ const std::map<std::string, std::string> Parser::loadInput(std::istream &jsonFil
 	while(getline(jsonFile, line)){
 		data += line;
 	}
+
+	if(data.substr(0, 1)!="{")
+		throw "No '{' at the beginning of the input.\n";
+
+	if(data.substr(data.length()-1, 1)!="}")
+		throw "No '}' at the end of the input.\n";
```

Comment:
1) Pl. a data[0] egyszerűbbnek tűnik egy karakter ellenőrzéséhez.
2) Nagyon szigorú vagy a karakterek pozíciójával kapcsolatban. Egy JSON fájlban elég szabadon lehet fehér karaktereket írni, amitől nem lesz hibás a fájl (pl. a } után is lehet tenni tetszőleges számút).